### PR TITLE
feat/material

### PR DESCRIPTION
**Pull request by joseglego**
# Changes
- Include [active_material](https://github.com/vigetlabs/active_material) 
- Define custom Material style for [xdan jQuery DateTimePicker](https://github.com/xdan/datetimepicker)
- Define custom Material style for `toggle_bool`
- Include generator argument `theme` to be able to install addons with/without material theme.
- Allow customization for `active_material`, `date_picker` & `toggle_bool` 
- Include Documentation

screencapture-localhost-3000-admin-login-2018-07-06-15_39_02

screencapture-localhost-3000-admin-categories-2018-07-06-14_05_51

screen shot 2018-07-06 at 15 35 43

screen shot 2018-07-06 at 16 31 30

### CODE REVIEW

File: spec/dummy/app/assets/stylesheets/active_admin.scss [hunk 1]
```diff
@@ -1,4 +1,5 @@
-//= require activeadmin_addons/all
+@import "activeadmin_addons/all";
+@import "activeadmin_addons/material";
```
Comment (monkeyci) [inline spec/dummy/app/assets/stylesheets/active_admin.scss]: Expected single quotes               string-quotes

File: spec/dummy/app/assets/stylesheets/active_admin.scss [hunk 2]
```diff
@@ -1,4 +1,5 @@
-//= require activeadmin_addons/all
+@import "activeadmin_addons/all";
```
Comment (monkeyci) [inline spec/dummy/app/assets/stylesheets/active_admin.scss]: Expected single quotes               string-quotes

File: lib/activeadmin_addons/engine.rb [hunk 3]
```diff
@@ -10,7 +10,7 @@ class Engine < ::Rails::Engine
         require_rel "support"
         require_rel "addons"
         require_rel "active_admin_config"
-        app.config.assets.precompile += %w(select.scss fileicons/*.png switches/switch_*.png)
+        app.config.assets.precompile += %w(select.scss fileicons/*.png switches/switch_*.png material/icons *.svg)
```
Comment (monkeyci) [inline lib/activeadmin_addons/engine.rb]: Line is too long. [114/100]

File: app/assets/stylesheets/activeadmin_addons/material.scss [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  }
+  .select2-selection__arrow {
+    height: 40px;
+  }
+}
+
+// Step 1: Date Range Filters
+.input.date_range {
+  input.datepicker {
+    background: #fff asset-url("active_material/icons/event.svg") 10px no-repeat;
```
Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/material.scss]: Expected single quotes             string-quotes

File: app/assets/stylesheets/activeadmin_addons/material.scss [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+@import './addons/material-datepicker';
+
+// Step 0: Select
+.select2-container .select2-selection--single {
+  height: 42px;
+
+  .select2-selection__rendered,  {
+    line-height: 42px;
+  }
+  .select2-selection__arrow {
```
Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/material.scss]: Expected empty line before rule    rule-empty-line-before

File: app/assets/stylesheets/activeadmin_addons/material.scss [hunk 6]
```diff
@@ -0,0 +1,22 @@
+@import "active_material";
+@import './addons/material-datepicker';
+
+// Step 0: Select
+.select2-container .select2-selection--single {
+  height: 42px;
+
+  .select2-selection__rendered,  {
```
Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/material.scss]: Expected single space before "{"   block-opening-brace-space-before

File: app/assets/stylesheets/activeadmin_addons/material.scss [hunk 7]
```diff
@@ -0,0 +1,22 @@
+@import "active_material";
```
Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/material.scss]: Expected single quotes             string-quotes

File: app/assets/stylesheets/activeadmin_addons/addons/material-datepicker.scss [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+  .xdsoft_timepicker .xdsoft_time_box>div>div:hover {
+    color: $dt-primary-color !important;
+    background: $dt-bg-border-color !important;
+  }
+
+  .xdsoft_calendar td.xdsoft_today {
+    &,
+    &.xdsoft_current {
+      background: $dt-selected-bg-color !important;
+      color: $dt-selected-color !important;
```
Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/addons/material-datepicker.scss]: Unexpected !important   declaration-no-important

File: app/assets/stylesheets/activeadmin_addons/addons/material-datepicker.scss [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+  .xdsoft_calendar td:hover,
+  .xdsoft_timepicker .xdsoft_time_box>div>div:hover {
+    color: $dt-primary-color !important;
+    background: $dt-bg-border-color !important;
+  }
+
+  .xdsoft_calendar td.xdsoft_today {
+    &,
+    &.xdsoft_current {
+      background: $dt-selected-bg-color !important;
```
Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/addons/material-datepicker.scss]: Unexpected !important   declaration-no-important

File: app/assets/stylesheets/activeadmin_addons/addons/material-datepicker.scss [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+  .xdsoft_timepicker .xdsoft_time_box>div>div.xdsoft_current {
+    color: $dt-selected-bg-color;
+    background: $dt-bg;
+    box-shadow: unset;
+  }
+
+  .xdsoft_calendar td:hover,
+  .xdsoft_timepicker .xdsoft_time_box>div>div:hover {
+    color: $dt-primary-color !important;
+    background: $dt-bg-border-color !important;
```
Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/addons/material-datepicker.scss]: Unexpected !important   declaration-no-important

File: app/assets/stylesheets/activeadmin_addons/addons/material-datepicker.scss [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+  .xdsoft_calendar td.xdsoft_current,
+  .xdsoft_timepicker .xdsoft_time_box>div>div.xdsoft_current {
+    color: $dt-selected-bg-color;
+    background: $dt-bg;
+    box-shadow: unset;
+  }
+
+  .xdsoft_calendar td:hover,
+  .xdsoft_timepicker .xdsoft_time_box>div>div:hover {
+    color: $dt-primary-color !important;
```
Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/addons/material-datepicker.scss]: Unexpected !important   declaration-no-important

File: lib/generators/activeadmin_addons/install/install_generator.rb [hunk 12]
```diff
@@ -2,33 +2,54 @@ module ActiveadminAddons
...
+        to_add
+      end
+
+      def css_assets
+        if options['theme'] == "material"
+          to_add = "$am-theme-primary: #342e48;\n"
+          to_add += "@import 'activeadmin_addons/all';\n"
+          to_add += "@import 'activeadmin_addons/material';\n"
+        else
+          to_add = "@import 'activeadmin_addons/all';\n"
```
Comment (monkeyci) [inline lib/generators/activeadmin_addons/install/install_generator.rb]: Useless assignment to variable - to_add.

File: lib/generators/activeadmin_addons/install/install_generator.rb [hunk 13]
```diff
@@ -2,33 +2,54 @@ module ActiveadminAddons
...
+        to_add = "#= require activeadmin_addons/all\n"
+        to_add += "#= require active_material\n" if options['theme'] == "material"
+        to_add
+      end
+
+      def css_assets
+        if options['theme'] == "material"
+          to_add = "$am-theme-primary: #342e48;\n"
+          to_add += "@import 'activeadmin_addons/all';\n"
+          to_add += "@import 'activeadmin_addons/material';\n"
```
Comment (monkeyci) [inline lib/generators/activeadmin_addons/install/install_generator.rb]: Useless assignment to variable - to_add.

File: lib/generators/activeadmin_addons/install/install_generator.rb [hunk 14]
```diff
@@ -2,33 +2,54 @@ module ActiveadminAddons
...
-          prepend_file("#{file_path}.scss", line_to_add)
-        rescue Errno::ENOENT
-          prepend_file("#{file_path}.css.scss", line_to_add)
+          prepend_file("#{file_path}.scss", css_assets)
+        rescue
+          prepend_file("#{file_path}.css.scss", css_assets)
+        end
+      end
+
+      private
```
Comment (monkeyci) [inline lib/generators/activeadmin_addons/install/install_generator.rb]: Keep a blank line before and after private.

**Comment by joseglego** [comment]
[[8p/23v] Reemplazar ActiveSkin con ActiveMaterial para que ActiveAdmin sea menos feo](https://trello.com/c/I7Frk8ox/461-8p-23v-reemplazar-activeskin-con-activematerial-para-que-activeadmin-sea-menos-feo)

File: app/assets/stylesheets/activeadmin_addons/addons/material-toggle_bool.scss [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+      left: -3px;
+      top: -3px;
+
+      transition: all .25s linear;
+      transform: translateX(0);
+    }
+
+    &.on {
+      background: $active-bg;
+      background-image: none;;
```
Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/addons/material-toggle_bool.scss]: Expected newline after ";" in a multi-line declaration block   declaration-block-semicolon-newline-after

File: app/assets/stylesheets/activeadmin_addons/addons/material-toggle_bool.scss [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+    background-image: none;
+    box-shadow: $box-shadow;
+
+    width: 36px;
+    height: 14px;
+    display: inline-block;
+    position: relative;
+    border-radius: 9px;
+
+
```
Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/addons/material-toggle_bool.scss]: Unexpected whitespace at end of line                           no-eol-whitespace

File: app/assets/stylesheets/activeadmin_addons/addons/material-toggle_bool.scss [hunk 17]
```diff
@@ -0,0 +1,53 @@
+$inactive-button: #aaa;
+$inactive-bg: #ddd;
+$active-button: $am-theme-primary;
+$active-bg: rgba($active-button, .6);
+
+$box-shadow: 0 2px 1px -1px rgba(0,0,0,.2), 0 1px 1px 0 rgba(0,0,0,.14), 0 1px 3px 0 rgba(0,0,0,.12);
```
Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/addons/material-toggle_bool.scss]: Expected single space after "," in a single-line function      function-comma-space-after

File: app/assets/stylesheets/activeadmin_addons/addons/material-toggle_bool.scss [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+    background-image: none;
+    box-shadow: $box-shadow;
+
+    width: 36px;
+    height: 14px;
+    display: inline-block;
+    position: relative;
+    border-radius: 9px;
+    
+    &::before {
```
Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/addons/material-toggle_bool.scss]: Expected empty line before rule        rule-empty-line-before

File: app/assets/stylesheets/activeadmin_addons/addons/material-toggle_bool.scss [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+    background: $inactive-bg;
+    background-image: none;
+    box-shadow: $box-shadow;
+
+    width: 36px;
+    height: 14px;
+    display: inline-block;
+    position: relative;
+    border-radius: 9px;
+
```
Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/addons/material-toggle_bool.scss]: Unexpected whitespace at end of line   no-eol-whitespace

File: app/assets/stylesheets/activeadmin_addons/material.scss [hunk 20]
```diff
@@ -0,0 +1,50 @@
...
+// Date Range
+.input.date_range {
+  input.datepicker {
+    background: #fff asset-url('active_material/icons/event.svg') 10px no-repeat;
+    padding-left: 40px;
+  }
+}
+
+// Numeric Range Filter
+.filter_numeric_range_filter  {
```
Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/material.scss]: Expected single space before "{"   block-opening-brace-space-before

File: app/assets/stylesheets/activeadmin_addons/material.scss [hunk 21]
```diff
@@ -0,0 +1,50 @@
+@import 'active_material';
+@import './addons/material-datepicker';
+@import './addons/material-toggle_bool';
+
+// Select
+.select2.select2-container.select2-container--default {
+  width: 100% !important;
```
Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/material.scss]: Unexpected !important              declaration-no-important

Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/material.scss]: Unexpected !important   declaration-no-important

File: app/assets/stylesheets/activeadmin_addons/material.scss [hunk 22]
```diff
@@ -0,0 +1,53 @@
+@import 'active_material';
+@import './addons/material-datepicker';
+@import './addons/material-toggle_bool';
+
+// Select
+.filter_select,
+.search_select_filter  {
+  .select2.select2-container.select2-container--default {
+    width: 100% !important;
```
Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/material.scss]: Unexpected !important              declaration-no-important

File: app/assets/stylesheets/activeadmin_addons/material.scss [hunk 23]
```diff
@@ -0,0 +1,53 @@
+@import 'active_material';
+@import './addons/material-datepicker';
+@import './addons/material-toggle_bool';
+
+// Select
+.filter_select,
+.search_select_filter  {
```
Comment (monkeyci) [inline app/assets/stylesheets/activeadmin_addons/material.scss]: Expected single space before "{"   block-opening-brace-space-before

**Pull request merged**